synapse/lib/cell.py

```diff
@@ -1177,16 +1177,22 @@ async def __anit__(self, dirn, conf=None, readonly=False, parent=None):
         self.cellinfo = await node.dict()
         self.onfini(node)
 
-        node = await self.hive.open(('cellvers',))
-        self.cellvers = await node.dict(nexs=True)
+        # Check the cell version didn't regress
+        if (lastver := self.cellinfo.get('cell:version')) is not None and self.VERSION < lastver:
+            mesg = f'Cell version mismatch for {self.getCellType()}. Current version ({self.VERSION}) is less than previous version ({lastver}).'
```

Comment:
Lets call this out specifically as a disallowed "version regression" or something rather than just a mismatch. Maybe something like: `{self.getCellType()} version regression is not allowed! Stored version: {lastver} Current version: {self.VERSION}`

Reply:
Also, we should probably explicitly make a `logger.error` call here too just to be sure it's really loud :)